src/journal/managed-journal-file.c

```diff
@@ -495,8 +495,8 @@ int managed_journal_file_open_reliably(
 
```

Comment:
this commit looks strange to me. i.e. shoehorning this into a function called "open" is wrong. apparently you don't want to open the file in this case, so it shouldn't be in a function called "open".

Hence, split out the stuff you actually are looking for into one or more new functions, and then call those from managed_journal_file_open_reliably(), but don't add "magic" flags to this funciton that does stuff in some magic conditions or does not.

Reply:
Yes I was pondering that too but I finally decided to go with the "magic" flags because it was already the case: passing  O_RDONLY prevents `managed_journal_file_open_reliably()` from disposing corrupted journals. That is also the case if O_CREAT is not passed.

The patch only changed the meaning of O_CREAT: it is used by the function to decide whether to create a new file if the old one was corrupted and disposed. It seemed that this new meaning made more sense actually.

Reply:
I think it's fine as is. I wanted to write that that the function should be renamed, e.g. from `_open_reliably(`) to `_rename_or_open_reliably()` or something like that, but this doesn't seem needed. After all, `open()` without `O_CREAT` will not create a file either and will return an error. So I think it's fine to make this conditional here. But please add a comment at the top of the function that describes the logic.